.cascade-code/Chapman.edu/_components/left_nav/left_nav.vtl

```diff
@@ -0,0 +1,153 @@
+## Formerly LeftNav_Velocity. Can be used with index and non-index pages.
+## New left nav format that works without a block
+
+#macro ( setVars )
+    ## Parent depends on whether page is an index or not
+    #set ( $isIndex = $currentPage.name == 'index' )
+    #if ( $isIndex )
+        #set ( $parent = $currentPage.parentFolder )
+        #set ( $parentDir = $parent.parentFolder )     
+    #else
+        #set ( $parentDir = $currentPage.parentFolder )
+    #end
+#end
+
+## Builds left navigation
+#macro ( buildLeftNav )
+    #buildLeftNavTitle()
+    <ul>
+    #set ( $siblings = $parentDir.children )
+    #buildLeftNavMenu($siblings)
+    </ul>
+#end
+
+## Builds the header for the left nav
+#macro ( buildLeftNavTitle )
+    #set ( $folderIndexPage = $_.locatePage("${parentDir.path}/index" ) )
+    
+    <div class="leftTitle">
+        <ul>
+            <li class="title">
+                <a href="${folderIndexPage.link}" target="_top" title="${_EscapeTool.xml($folderIndexPage.label)}" >
+                    ${_EscapeTool.xml($folderIndexPage.label)}
+                </a>
+            </li>
+        </ul>
+    </div>
+#end
+
+## Builds menu in the left nav
+#macro ( buildLeftNavMenu $siblings )
+    #foreach ( $sibling in $siblings )
+        #set ( $isVisible = $sibling.metadata.getDynamicField('Hide from navigation').value != "Yes" )
+
+        ## Sibling is a single page
+        #if ( $sibling.assetType == 'page' && $isVisible && ($sibling.name != 'index') )
+            #buildSiblingPageListItem($sibling)
+
+        ## Sibling is a folder
+        #elseif ( $sibling.assetType == 'folder' && $isVisible )
+            #set ( $indexPage = $_.locatePage("${sibling.path}/index") )
+            #set ( $visibleIndex = $indexPage.metadata.getDynamicField('Hide from navigation').value != "Yes" )
+            
+            ## Only if the index page is also visible
+            #if ( $visibleIndex )
+                #buildSiblingFolderListItem($sibling $indexPage)
+            #end
+        #end
+    #end
+#end
```

Comment:
The foreach feels out of place here. The following parses a little more easily for me:

```
## Builds left navigation
#macro ( buildLeftNav )
    ## Data
    #set ( $siblings = $parentDir.children )

    ## Output
    #buildLeftNavTitle()
    
    <ul>
    #foreach ( $sibling in $siblings )
      #buildLeftNavItem($sibling)
    #end
    </ul>
#end
```

Then basically rename this macro and remove of the foreach here.

Alternately, keep this macro, but move the `ul` into it:

```
## Builds left navigation
#macro ( buildLeftNav )
    ## Data
    #set ( $siblings = $parentDir.children )

    ## Output
    #buildLeftNavTitle()
    #buildLeftNavMenu($siblings)
#end
```

I think I like that best. (Hopefully, this isn't what you originally had.)